internal/receiver/prometheusremotewritereceiver/client.go

```diff
@@ -65,8 +66,17 @@ func (prwc *MockPrwClient) SendWriteRequest(wr *prompb.WriteRequest) error {
 
 	ctx, cancel := context.WithTimeout(context.Background(), prwc.Timeout)
 	defer cancel()
-
-	err = prwc.Client.Store(ctx, compressed)
+	retry := 3
+	for retry > 0 {
+		err = prwc.Client.Store(ctx, compressed)
+		if nil == err {
+			break
+		}
+		if strings.Contains(err.Error(), "connection refused") {
```

Comment:
maybe something like
```suggestion
		if errors.Is(err, syscall.ECONNREFUSED) {
```

also should we silently and immediately retry if not?